src/serviceconnector-passwordless/azext_serviceconnector_passwordless/_utils.py

```diff
@@ -48,6 +50,34 @@ def run_cli_cmd(cmd, retry=0, interval=0, should_retry_func=None, should_return_
         raise e
 
 
+def run_cli_cmd_base(cmd, retry=0, interval=0, should_retry_func=None):
+    '''Run a CLI command
+    :param cmd: The CLI command to be executed
+    :param retry: The times to re-try
+    :param interval: The seconds wait before retry
+    '''
+    output = _in_process_execute(cmd)
+
+    if output.error or (should_retry_func and should_retry_func(output)):
+        if retry:
+            time.sleep(interval)
+            return run_cli_cmd(cmd, retry - 1, interval)
+        raise CLIInternalError('Command execution failed, command is: '
+                               '{}, error message is: \n {}'.format(cmd, output.error))
+    return output.result
+
+
+def _in_process_execute(command):
+    import shlex
+
+    if command.startswith('az '):
+        command = command[3:]
+
+    cli = get_default_cli()
+    cli.invoke(shlex.split(command), out_file=open(os.devnull, 'w'))  # Don't print output
```

Comment:
As the doc states:  The shlex module is only designed for Unix shells. https://docs.python.org/3/library/shlex.html#shlex.split
It does not work well in windows when the split str has file absolute path in it, like `C:\Program Files\Git\cmd\git.EXE --version`
It is recommended to construct args as list from the beginning.